source/common/unified_memory_allocation/include/uma/memory_pool_ops.h

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+
+#ifndef UMA_MEMORY_POOL_OPS_H
+#define UMA_MEMORY_POOL_OPS_H 1
+
+#include <uma/base.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// TODO: create macro for intializing this structure (including version fields)
+struct uma_memory_pool_ops_t {
+  uint64_t version_major;
```

Comment:
you could create a base struct with versions and use it both here and in the provider.

Reply:
I've replaced it with a single variable and a macro to encode major/minor.